### ISSUE TITLE
Gradle kotlin plugins 1.5.30-RC cause spotless to hang

### ISSUE THREAD

**Issue by nedtwigg** [diffplug/spotless#918]
I don't have a public test case, but reporting here in case anyone else bumps into this. I have a big multimodule project which works when my `settings.gradle` is like this:

```gradle
// settings.gradle
pluginManagement { plugins {
  id 'org.jetbrains.kotlin.jvm'                  version '1.5.21'
  id 'org.jetbrains.kotlin.kapt'                 version '1.5.21'
  id 'org.jetbrains.kotlin.plugin.serialization' version '1.5.21'
  ... (other plugins)
```

I updated to `1.5.30-RC`, and now Spotless hangs indefinitely. Very strange, should not be related at all, but I have isolated it to only this change.

Also strange, is that some Spotless tasks hang, and others don't. Specifically, either of these two tasks will hang in the root project, but will not hang in any of the child projects.

```gradle
spotless {
		format 'misc', {
			target '*.md', '.gitignore', '.gitattributes'
			indentWithTabs()
			trimTrailingWhitespace()
			endWithNewline()
		}
		groovyGradle {
			if (project == rootProject) {
				target '*.gradle', 'gradle/*.gradle'
			} else {
				target '*.gradle'
			}
			greclipse()
		}
```

To summarize: something changed in the plugins from `1.5.21` to `1.5.30-RC` which causes spotless tasks in the root project to hang indefinitely, but does not affect spotless tasks elsewhere.

🤷

Not gonna put a ton of effort into investigating this for now. Bug is present as of Gradle 7.2 and Spotless 5.14.3.

**Comment by evertsmits** [comment]
Hi @nedtwigg I am going to respond here as this issue was originally mentioned by me. I've encountered this issue when updating to Kotlin 1.5.21. 

    kotlin {
        target '**/*.kt'
        targetExclude "caches/**", "wrapper/**"
        ktlint('0.41.0')
    }
    
My :spotlessKotlin task would hang. It turns out that all the spotless tasks in my spotless.gradle keep hanging indefinitely while using Kotin 1.5.21 or newer. No issues with Kotlin 1.5.10 (without making a single change to my spotless.gradle file)

**Comment by nedtwigg** [comment]
Very interesting that my build is okay in `1.5.21` but broken in `1.5.30`, whereas yours is broken even in `1.5.21`. I'm guessing this is going to be very difficult to debug, and it's easy for me to workaround, so I don't expect a fix anytime soon. If anybody finds any diagnostic info on what's going on, please share!

**Comment by evertsmits** [comment]
@nedtwigg Would you know of any alternative to use for formatting purposes? Because I have a feeling that we will be stuck on kotlin 1.5.10 for a long time. We prefer not to turn off any formatting tools as this always results in unnecessary pull request comments regarding formatting etc.

**Comment by jbduncan** [comment]
@evertsmits You can try following the [ktlint homepage](tps://ktlint.github.io/)'s instructions on setting up some custom ktlint tasks in your Gradle project, instead of using Spotless, and see if that improves things? My team uses this solution successfully, but I can't remember for sure if we're on the latest Kotlin.

If that doesn't work, then the only other thing I can think of is to try ktfmt, which formats Kotlin code quite differently to ktlint.

It's not clear to me from what little info there is why Spotless hangs... 🤔

**Comment by evertsmits** [comment]
@jbduncan Thanks for sharing the suggestions. I've already started implementing ktlint instead of Spotless. I'm sorry I could not provide anymore info than this. Trying to run ./gradlew spotlessApply with --scan does not give me any more info other than that it keeps on running:
![image (14)](https://user-images.githubusercontent.com/31696159/132396658-16537b99-b322-4e64-9f9a-d2babf989847.png)

**Comment by nedtwigg** [comment]
Interesting! Spotless 6.0.0 makes this problem *worse* for me. But, Gradle 7.3-rc-5 fixes it completely, for both Spotless 5.x and 6.0. Don't know the root cause, but Gradle 7.3 fixes it.
